cognite/client/data_classes/datapoints.py

```diff
@@ -79,6 +87,47 @@ def numpy_dtype_fix(element: np.float64 | str) -> float | str:
         raise
 
 
+_NOT_SET = object()
+
+
+@dataclass(frozen=True)
+class DatapointsQuery:
+    """Represent a user request for datapoints for a single time series"""
+
+    id: InitVar[int | None] = None
+    external_id: InitVar[str | None] = None
+    start: int | str | datetime = _NOT_SET  # type: ignore [assignment]
```

Comment:
I need to know if something was passed or not by the user, and `None` is perfectly valid input for many of the arguments.

If we allow `type[_NOT_SET]`, it will most likely become a huge source of confusion for our users - I think 😅